components/navigation/MobileNavbar.tsx

```diff
@@ -8,9 +7,57 @@ import SearchRouteIcon from '../icons/navigation/SearchRouteIcon';
 import AboutRouteIcon from '../icons/navigation/AboutRouteIcon';
 import AddItemRouteIcon from '../icons/navigation/AddItemRouteIcon';
 import MessageRouteIcon from '../icons/navigation/MessageRouteIcon';
+import NotificationDot from '../NotificationDot';
+
+import { usePathname } from 'next/navigation';
+import { useEffect, useState } from 'react';
+import selectUserUnreadConversations from '@/supabase/models/messaging/selectUserUnreadConversations';
+import selectLoggedUserId from '@/supabase/utils/selectLoggedUserId';
+import notificationWatcher from '@/supabase/channels/notificationsWatcher';
 
 const MobileNavbar = () => {
   const pathname = usePathname();
+  const [userId, setUserId] = useState<string>('');
+  const [hasNotification, setHasNotification] = useState<boolean>(false);
+
+  useEffect(() => {
+    const getUserId = async () => {
+      try {
+        const data = await selectLoggedUserId();
+        if (!data) return;
+        setUserId(data);
+      } catch (error) {
+        console.error('Error fetching user Id:', error);
+      }
+    };
+    getUserId();
+  }, []);
+
+  useEffect(() => {
+    const getUnreadConversations = async () => {
+      if (!userId) return;
```

Comment:
```ts
if (!userId || pathname === '/conversations') {
  setHasNotification(false)
  return
}
```

we can just have an early return and avoid a database call entirely if we are already on './conversations'

Reply:
Yes this works fine just had to add `pathname` to rerun the `useEffect`